packages/react-reconciler/src/ReactFiberThrow.new.js

```diff
@@ -507,6 +508,14 @@ function throwException(
           root,
           rootRenderLanes,
         );
+
+        // Even though the user may not be affected by this error, we should
+        // still log it so it can be fixed.
+        // TODO: For now, we only log errors that occur during hydration, but we
+        // probably want to log any error that is recovered from without
+        // triggering an error boundary — or maybe even those, too. Need to
+        // figure out the right API.
+        logRecoverableError(root.errorLoggingConfig, value);
```

Comment:
I think we probably don't want to log errors to this if they also end up surfacing a user error afterwards. It seems like kind of a misnomer in the API if it also gets passed errors that end up erroring. Even though you could maybe try to dedupe them this seems tricky to do in user space. The idle callback (if we keep it) also makes this even trickier since the error boundaries are not idle.

Reply:
I agree but I can't think of a reliable way to implement it that doesn't fail in some edge case. Once it falls back to client rendering, we can't know for sure if a subsequent error is the same one or a different one. So here I erred on the side of logging too much.

To argue for the other approach, you could say that when that situation does happen, it's ok that the first error gets unintentionally silenced because it will be unsilenced after the second error is fixed.